### DISCUSSION TITLE
깊은 복사(deep copy), 얕은 복사(shallow copy)에 대해서 설명해주세요

### DISCUSSION THREAD

**Discussion by kunsanglee** [2024-woowacourse-study/level-interview#63] [Java]
.

**Comment by J-I-H-O** [comment]
먼저 얕은 복사는 객체의 참조를 복사하는 것을 말합니다. 따라서 복사된 객체와 원본 객체가 동일한 참조를 공유하고, 두 객체 중 하나의 변경이 다른 객체에도 영향을 줄 수 있습니다.

반면 깊은 복사는 주소값이 아닌 실제 값을 통째로 복사하는 것을 말합니다. 원본 객체와 복사된 객체는 완전히 독립적인 객체로, 서로 영향을 주지 않습니다.

(꼬리질문) 자바에서 깊은 복사를 구현하는 방법?

**Reply by seunghye218** [reply]
자바에서 깊은 복사를 구현하는 방법?

**Comment by 3Juhwan** [comment]
자바에서 얕은 복사는 객체를 복사할 때 해당 객체의 필드들을 단순히 복사하는 것이며, 복사된 객체와 원본 객체는 같은 참조를 공유합니다. 반면에 깊은 복사는 객체의 필드들을 재귀적으로 모두 복사하여 복사본과 원본이 완전히 독립된 객체가 되도록 합니다. 이로써 깊은 복사는 객체 간의 의존성을 제거하여 데이터 무결성을 보장하고 예상치 못한 부작용을 방지하는 데 도움을 줍니다.

**Comment by GIVEN53** [comment]
앝복
객체 참조 복사
동일한 참조
영향 전파

깊복
객체 값 복사
독립된 객체
영향  x, 데이터 무결성
